Build fails

See https://github.com/adafruit/Adafruit_Python_PlatformDetect/actions/runs/4047615129/jobs/6961847839. It might be an issue of running Python 3.11 in GitHub Actions.

Output:

```
Run find -type f -not -path "./.*" -not -path "./docs*" \( -name "*.py" -o -name "*.toml" \) -exec sed -i -e "s/0.0.0+auto.0/3.40.0/" {} +
  find -type f -not -path "./.*" -not -path "./docs*" \( -name "*.py" -o -name "*.toml" \) -exec sed -i -e "s/0.0.0+auto.0/3.40.0/" {} +
  python -m build
  twine upload dist/*
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/[3](https://github.com/adafruit/Adafruit_Python_PlatformDetect/actions/runs/4047615129/jobs/6961847839#step:6:3).11.1/x6[4](https://github.com/adafruit/Adafruit_Python_PlatformDetect/actions/runs/4047615129/jobs/6961847839#step:6:4)
    TWINE_USERNAME: ***
    TWINE_PASSWORD: ***
/opt/hostedtoolcache/Python/3.11.1/x[6](https://github.com/adafruit/Adafruit_Python_PlatformDetect/actions/runs/4047615129/jobs/6961847839#step:6:6)4/bin/python: No module named build
Error: Process completed with exit code 1.
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -19,18 +19,18 @@ jobs:
         tr '_' '-'
         )
     - name: Set up Python 3.7
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v4
       with:
         python-version: 3.7
     - name: Versions
       run: |
         python3 --version
     - name: Checkout Current Repo
-      uses: actions/checkout@v1
+      uses: actions/checkout@v3
       with:
         submodules: true
     - name: Checkout tools repo
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3
       with:
         repository: adafruit/actions-ci-circuitpython-libs
         path: actions-ci

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -12,20 +12,20 @@ jobs:
   upload-pypi:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v3
     - name: Check For pyproject.toml
       id: need-pypi
       run: |
         echo pyproject-toml=$( find . -wholename './pyproject.toml' ) >> $GITHUB_OUTPUT
     - name: Set up Python
       if: contains(steps.need-pypi.outputs.pyproject-toml, 'pyproject.toml')
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v4
       with:
         python-version: '3.x'
     - name: Install dependencies
-      if: contains(steps.need-pypi.outputs.pyproject.toml, 'pyproject.toml')
+      if: contains(steps.need-pypi.outputs.pyproject-toml, 'pyproject.toml')
       run: |
-        python -m pip install --upgrade pip build
+        python -m pip install --upgrade pip
         pip install --upgrade build twine
     - name: Build and publish
       if: contains(steps.need-pypi.outputs.pyproject-toml, 'pyproject.toml')

